chapter_preliminaries/tablesaw.ipynb

```diff
@@ -0,0 +1,341 @@
+    "A row whose alley type is \"Pave\" will set values of \"Alley_Pave\" and \"Alley_nan\" to true and false.\n",
+    "A row with a missing alley type will set their values to false and true.\n",
+    "After this, we will add these columns to the original data/table but converting them to double so it changes true and false to 1 and 0 respectively. Finally, we remove the original column \"Alley\"."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       " NumRooms  |  Alley_Pave  |  Alley_nan  |\n",
+       "-----------------------------------------\n",
+       "        3  |           1  |          0  |\n",
+       "        2  |           0  |          1  |\n",
+       "        4  |           0  |          1  |\n",
+       "        3  |           0  |          1  |"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "StringColumn col = (StringColumn) inputs.column(\"Alley\");\n",
+    "List<BooleanColumn> dummies = col.getDummies();\n",
+    "inputs.removeColumns(col);\n",
+    "inputs.addColumns(DoubleColumn.create(\"Alley_Pave\", dummies.get(0).asDoubleArray()), \n",
+    "                  DoubleColumn.create(\"Alley_nan\", dummies.get(1).asDoubleArray())\n",
+    "                 );\n",
+    "inputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "origin_pos": 8
+   },
+   "source": [
+    "## Conversion to the Tensor Format\n",
+    "\n",
+    "Now that all the entries in `inputs` and `outputs` are numerical, they can be converted to the `NDArray` format.\n",
+    "Once data are in this format, they can be further manipulated with those NDArray functionalities that we have introduced in :numref:`sec_NDArray`.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%maven ai.djl:api:0.9.0\n",
+    "%maven ai.djl:basicdataset:0.9.0\n",
+    "\n",
+    "// See https://github.com/awslabs/djl/blob/master/mxnet/mxnet-engine/README.md\n",
+    "// MXNet \n",
+    "%maven ai.djl.mxnet:mxnet-engine:0.9.0\n",
+    "%maven ai.djl.mxnet:mxnet-native-auto:1.7.0-backport"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "ND: (4, 3) cpu() int32\n",
+       "[[ 3,  1,  0],\n",
+       " [ 2,  0,  1],\n",
+       " [ 4,  0,  1],\n",
+       " [ 3,  0,  1],\n",
+       "]\n"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import ai.djl.ndarray.*;\n",
+    "NDManager nd = NDManager.newBaseManager();\n",
+    "NDArray x = nd.create(inputs.as().intMatrix());\n",
```

Comment:
the dtype of original book is float64. use floatMatrix() or doubleMatrix() instead

Reply:
True, done.